src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/MessageBody.cs

```diff
@@ -215,7 +213,24 @@ private void TryProduceContinue()
             }
         }
 
-        protected abstract bool Read(ReadableBuffer readableBuffer, WritableBuffer writableBuffer, out ReadCursor consumed, out ReadCursor examined);
+        private void TryInit()
+        {
+            if (!_context.HasStartedConsumingRequestBody)
+            {
+                _context.HasStartedConsumingRequestBody = true;
+                OnReadStart();
```

Comment:
Start was cleaner tbh. I understand why it was removed but it meant we avoid this lazy initialize once logic.